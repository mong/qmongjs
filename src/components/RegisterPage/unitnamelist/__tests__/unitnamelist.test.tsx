import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import { QueryParamProvider } from "use-query-params";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import { UnitNameList } from "../";
import namesData from "../../../../dev-tools/data/unitnames.json";

it("unit name selection renders", async () => {
  const { container } = render(
    <BrowserRouter>
      <QueryParamProvider ReactRouterRoute={Route}>
        <UnitNameList
          nestedUnitNames={namesData.nestedUnitNames}
          treatment_units={[""]}
          update_treatment_units={() => {}}
        />
      </QueryParamProvider>
    </BrowserRouter>
  );
  expect(container.getElementsByClassName("tu_list")[0]).toHaveStyle(
    "display: none"
  );
  expect(container).toMatchSnapshot();

  act(() => {
    userEvent.click(screen.getByText(/Vis alle/i));
  });
  expect(container.getElementsByClassName("tu_list")[0]).not.toHaveStyle(
    "display: none"
  );
});

it("should close the list of TUs when ESC/ESCAPE is clicked", () => {
  const { container } = render(
    <BrowserRouter>
      <QueryParamProvider ReactRouterRoute={Route}>
        <UnitNameList
          nestedUnitNames={namesData.nestedUnitNames}
          treatment_units={[""]}
          update_treatment_units={() => {}}
        />
      </QueryParamProvider>
    </BrowserRouter>
  );
  act(() => {
    userEvent.click(screen.getByText(/Vis alle/i));
  });
  expect(container.getElementsByClassName("tu_list")[0]).not.toHaveStyle(
    "display: none"
  );
  act(() => {
    var event = new KeyboardEvent("keydown", { key: "Esc" });
    global.dispatchEvent(event);
  });
  expect(container.getElementsByClassName("tu_list")[0]).toHaveStyle(
    "display: none"
  );
});
