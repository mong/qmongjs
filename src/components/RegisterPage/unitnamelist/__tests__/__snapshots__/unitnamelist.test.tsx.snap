// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit name selection renders 1`] = `
<div>
  <div
    class="hospital_list_btn_container"
  >
    <button
      class="hospital_list_btn"
    >
      Vis alle
    </button>
  </div>
  <div
    class="tu_list"
    style="display: none;"
    test-id="tu_list"
  >
    <h2
      class="tu_list_header"
    >
      Velg behandlingsenheter (maks 5)
    </h2>
    <div
      class="all_tu"
    >
      <div
        class="tu_list_rhf"
      >
        <h3
          class="tu_list_rhf_header"
        >
          Helse Midt-Norge RHF
        </h3>
        <div
          class="tu_list_rhf_content"
        >
          <div
            class="tu_list_hfs_in_rhf"
          >
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Møre og Romsdal HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Kristiansund
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Molde
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volda
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Ålesund
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Nord-Trøndelag HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Levanger
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Namsos
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  St. Olavs Hospital HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Orkdal
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Røros
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  St. Olavs
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tu_list_rhf"
      >
        <h3
          class="tu_list_rhf_header"
        >
          Helse Nord RHF
        </h3>
        <div
          class="tu_list_rhf_content"
        >
          <div
            class="tu_list_hfs_in_rhf"
          >
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Finnmarkssykehuset HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Hammerfest
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kirkenes
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helgelandssykehuset HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Mo i Rana
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Mosjøen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Sandnessjøen
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Nordlandssykehuset HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Bodø
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Lofoten
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Vesterålen
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Universitetssykehuset Nord-Norge HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Harstad
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Narvik
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Tromsø
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tu_list_rhf"
      >
        <h3
          class="tu_list_rhf_header"
        >
          Helse Sør-Øst RHF
        </h3>
        <div
          class="tu_list_rhf_content"
        >
          <div
            class="tu_list_hfs_in_rhf"
          >
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Akershus universitetssykehus HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Ahus
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kongsvinger
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Oslo universitetssykehus HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Aker
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Radiumhospitalet
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Rikshospitalet
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Ullevål
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Private ideelle sykehus
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Diakonhjemmet
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Lovisenberg
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Martina Hansens hospital
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sunnaas sykehus HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Sunnaas
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sykehuset Innlandet HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Elverum
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Gjøvik
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Hamar
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Lillehammer
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Tynset
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sykehuset Telemark HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Kragerø
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Notodden
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Porsgrunn
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Skien
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sykehuset i Vestfold HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Larvik
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Tønsberg
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sykehuset Østfold HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Askim
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kalnes
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Moss
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Sørlandet sykehus HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Arendal
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Flekkefjord
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kristiansand
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Vestre Viken HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Bærum
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Drammen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kongsberg
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Ringerike
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tu_list_rhf"
      >
        <h3
          class="tu_list_rhf_header"
        >
          Helse Vest RHF
        </h3>
        <div
          class="tu_list_rhf_content"
        >
          <div
            class="tu_list_hfs_in_rhf"
          >
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Bergen HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Haukeland
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Kysthospitalet i Hagevik
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Voss
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Fonna HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Haugesund
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Odda
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Stord
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Førde HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Førde
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Lærdal
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Nordfjord
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Helse Stavanger HF
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Stavanger
                </button>
              </div>
            </div>
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Private ideelle sykehus
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Betanien
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Haraldsplass
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Haugesund sanitet
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tu_list_rhf"
      >
        <h3
          class="tu_list_rhf_header"
        >
          Private
        </h3>
        <div
          class="tu_list_rhf_content"
        >
          <div
            class="tu_list_hfs_in_rhf"
          >
            <div
              class="tu_list_hf"
            >
              <div
                class="tu_list_hf_btn_constainer"
              >
                <button
                  class="tu_list_hf_btn"
                >
                  Private sykehus
                </button>
              </div>
              <div
                class="tu_list_hospital"
              >
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Agder
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Bergen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Bodø
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Colosseum Nobel
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Colosseum Stavanger
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Drammen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Frogner
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Hinna Park
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Marken
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Rosenborgklinikken
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Aleris Tromsø
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Ibsensykehuset Porsgrunn
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Idrettens helsesenter
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  LHL Feiring
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  LHL Gardermoen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Medi 3 Hobøl
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  NIMI
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Norsk Diabetessenter
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Oslofjordklinikken Sandnes
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Oslofjordklinikken Sandvika
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Privatsykehuset Haugesund
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Sarpsborg Spesialistsenter
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volvat
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volvat Fredrikstad
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volvat Majorstuen
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volvat Tromsø
                </button>
                <button
                  class="tu_list_hospital_btn"
                >
                  Volvat Ulriksdalen
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="tu_list_close_btn"
    >
      <i
        class="far fa-times-circle"
      />
    </button>
  </div>
</div>
`;
